Release 0.8

CPerezz opened the pull request.

CPerezz committed:
Update `dusk-poseidon` from v0.21.0 to v0.22

Resolves: #54

CPerezz committed:
Merge pull request #56 from dusk-network/poseidon_0.22

Update `dusk-poseidon` from v0.21.0 to v0.22

CPerezz committed:
Bump to 0.8.0-rc.0

ZER0 commented on .github/workflows/dusk_ci.yml:
```diff
@@ -12,7 +12,6 @@ jobs:
         with:
           profile: minimal
           toolchain: nightly
```
Remove also all the `toolchain` since they're useless.

vlopes11 replied:
Fixed

ZER0 commented on CHANGELOG.md:
```diff
@@ -7,26 +7,33 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
+
+### Changed
+
+- Change `PartialEq` and `Eq` impls for `PublicKey`
```
Don't we have an issue for this?

vlopes11 replied:
Fixed

ZER0 commented on src/keys/public.rs:
```diff
@@ -54,3 +64,46 @@ impl Serializable<32> for PublicKey {
+        let sk1 = PublicKey::from(&SecretKey::random(&mut OsRng));
+        let sk2 = PublicKey::from(&SecretKey::random(&mut OsRng));
+
+        assert_eq!(sk1, sk1);
```
What's the point to do this?

vlopes11 replied:
This is a simple assertion that the same key equals itself. Since the implementation of the equality is non-trivial (e.g. its a formula evaluation), its good to cover also the basic case.

ZER0 replied:
Ok. Please add `#[allow(clippy::eq_op)]` to the test then, and move it as integration test and not as unit test (you're not testing any internal here, you're testing public types' behavior).

ZER0 commented on src/keys/public.rs:
```diff
@@ -54,3 +64,46 @@ impl Serializable<32> for PublicKey {
+
+        // With all coordinates being different the points are the same ie.
+        // equalty holds using this technique.
+        let a = JubJubScalar::from(2u64);
+        let b = JubJubScalar::from(7u64);
+        let c = JubJubScalar::from(4u64);
+        let d = JubJubScalar::from(5u64);
+        let e = JubJubScalar::from(567758785u64);
```
Either group those variable in a tuple or add `#[allow(clippy::many_single_char_names)]` to the test.

vlopes11 replied:
Fixed

ZER0 reviewed (changes requested):
Few nits. It's unclear to me the issue related to the `PartialEq` since this is a release.

Also, before releasing, I would clean up a bit `dusk-pki`, so my suggestion is once you address the comment I will ask you to review a small PR to also be merged on `release-0.8` that remove some unnecessary things unrelated to the `PartialEq`.

vlopes11 committed:
Change `PartialEq` and `Eq` impls for `PublicKey`

Efficient `PartialEq` ECC implementation for extended points. Prevents
converting into `JubJubAffine` to evaluate the equality.

Resolves: #53

vlopes11 committed:
Bump to v0.8.0

ZER0 committed:
Remove unnecessary references

Resolves: #58

ZER0 committed:
Merge pull request #59 from dusk-network/clean-up

Remove unnecessary references
